tests/unit_tests/dft/include/test_common.hpp

```diff
@@ -0,0 +1,228 @@
+    if (!ok) {
+        std::cout << "relative error = " << rerr << " absolute error = " << aerr
+                  << " limit = " << bound << std::endl;
+    }
+    return ok;
+}
+
+template <typename fp>
+bool check_equal(fp x, fp x_ref, int error_mag, std::ostream &out) {
+    bool good = check_equal(x, x_ref, error_mag);
+
+    if (!good) {
+        out << "Difference in result: DPC++ " << x << " vs. Reference " << x_ref << std::endl;
+    }
+    return good;
+}
+
+template <typename fp>
+bool check_equal_vector(fp *v, fp *v_ref, int n, int inc, int error_mag, std::ostream &out) {
+    constexpr size_t max_print = 20;
+    int abs_inc = std::abs(inc), count = 0;
+    bool good = true;
+
+    for (int i = 0; i < n; i++) {
+        if (!check_equal(v[i * abs_inc], v_ref[i * abs_inc], error_mag)) {
+            int i_actual = (inc > 0) ? i : n - i;
+            std::cout << "Difference in entry " << i_actual << ": DPC++ " << v[i * abs_inc]
+                      << " vs. Reference " << v_ref[i * abs_inc] << std::endl;
+            good = false;
+            count++;
+            if (count > max_print) {
+                return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+template <typename vec1, typename vec2>
+bool check_equal_vector(vec1 &v, vec2 &v_ref, int n, int inc, int error_mag, std::ostream &out) {
+    constexpr size_t max_print = 20;
+    int abs_inc = std::abs(inc), count = 0;
+    bool good = true;
+
+    for (int i = 0; i < n; i++) {
+        if (!check_equal(v[i * abs_inc], v_ref[i * abs_inc], error_mag)) {
+            int i_actual = (inc > 0) ? i : n - i;
+            std::cout << "Difference in entry " << i_actual << ": DPC++ " << v[i * abs_inc]
+                      << " vs. Reference " << v_ref[i * abs_inc] << std::endl;
+            good = false;
+            count++;
+            if (count > max_print) {
+                return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+// Random initialization.
+template <typename fp>
+static fp rand_scalar() {
+    return fp(std::rand()) / fp(RAND_MAX) - fp(0.5);
+}
+template <typename fp>
+static std::complex<fp> rand_complex_scalar() {
+    return std::complex<fp>(rand_scalar<fp>(), rand_scalar<fp>());
+}
+template <>
+std::complex<float> rand_scalar() {
+    return rand_complex_scalar<float>();
+}
+template <>
+std::complex<double> rand_scalar() {
+    return rand_complex_scalar<double>();
+}
+
+template <>
+int32_t rand_scalar() {
+    return std::rand() % 256 - 128;
+}
+
+template <typename fp>
+void rand_vector(fp *v, int n, int inc) {
+    int abs_inc = std::abs(inc);
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
+
+template <typename vec>
+void rand_vector(vec &v, int n, int inc) {
+    using fp = typename vec::value_type;
+    int abs_inc = std::abs(inc);
+
+    v.resize(n * abs_inc);
+
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
```

Comment:
```suggestion
template <typename vec>
void rand_vector(vec &v, std::size_t n) {
    using fp = typename vec::value_type;

    v.resize(n);

    for (std::size_t i = 0; i < n; ++i) {
        v[i] = rand_scalar<fp>();
    }
}
```
The first impl wasn't used and inc was a always 1

Reply:
Done in [56d18e6](https://github.com/oneapi-src/oneMKL/pull/261/commits/56d18e60d45dc3a0b617eef414eb0e84cbe6d7c3).